### PR TITLE
Handle duplicate key error in LoggerExtractor

### LINKED ISSUE #727: Parametrize apiLoggerId with  splitApis configuration turned on breaks extraction
## Where did you encounter this bug?

* [x] Extractor
* [] Creator

## Describe the bug
Setup:
Two api in one ApiManagement instance. 
One Api has Diagnostics Logs setuped (application insights)
Parameters to run the extractor: 
```
"paramApiLoggerId": "true",
"splitApis": "true"
```
exception is raised using this configuration with the error: 

`System.ArgumentException: 'An item with the same key has already been added.
`
## What version of the DevOps Resource Kit is being used?

Main branch

## The configuration file being used?
```
{
    "sourceApimName": "apim1",
    "destinationApimName": "apim2",
    "resourceGroup": "resource-group",
    "fileFolder": "path\to\folder",
    "paramApiLoggerId": "true",
    "splitApis": "true"
}
```

## What logs (if any) were produced?

**An item with the same key has already been added.**

## Expected behavior
No exception

### PR DESCRIPTION
**Pull request by f-alizada**
Closes #727 
PR addresses the bug with covering this method by test.

### CODE REVIEW

File: src/ArmTemplates/Extractor/EntityExtractors/LoggerExtractor.cs [hunk 1]
```diff
@@ -148,9 +148,11 @@ public class LoggerExtractor : ILoggerExtractor
 
                 if (!diagnosticLoggerBindings.IsNullOrEmpty())
                 {
-                    this.Cache.ApiDiagnosticLoggerBindings.Add(
-                        NamingHelper.GenerateValidParameterName(curApiName, ParameterPrefix.Api), 
-                        diagnosticLoggerBindings);
+                    var diagnosticLoggerKey = NamingHelper.GenerateValidParameterName(curApiName, ParameterPrefix.Api);
```
[the remark this answers is not in the document]
Comment (f-alizada) [reply src/ArmTemplates/Extractor/EntityExtractors/LoggerExtractor.cs:151]: The conflict was not in the resource but in the key of the dictionary. 
If I got the question correctly. 
The method was called multiple times for different scenarios, and tried to put into the cache the same value

File: tests/ArmTemplates.Tests/Extractor/Scenarios/LoggerExtractorTests.cs [hunk 2]
```diff
@@ -70,5 +72,63 @@ public async Task GenerateLoggersTemplates_ProperlyLaysTheInformation()
...
+            }
+
+            // assert
+            loggerExtractor.Cache.ServiceLevelDiagnosticLoggerBindings.Count.Should().Be(1);
+            loggerExtractor.Cache.ApiDiagnosticLoggerBindings.Count.Should().Be(2);
+            
+            foreach( var (key, value) in expectedApiDiagnosticLoggerBingingsValues)
+            {
+                loggerExtractor.Cache.ApiDiagnosticLoggerBindings[key].Count.Should().Be(1);
+                loggerExtractor.Cache.ApiDiagnosticLoggerBindings[key].First().Equals(value.First()).Should().BeTrue();
```
[the remark this answers is not in the document]
Comment (f-alizada) [reply tests/ArmTemplates.Tests/Extractor/Scenarios/LoggerExtractorTests.cs:130]: The check is to verify that the objects are equal, based on the [class implementation](https://github.com/Azure/azure-api-management-devops-resource-kit/blob/ced2999486c5c3655ee61660b5d34b5f8351002f/src/ArmTemplates/Common/Templates/Logger/Cache/DiagnosticLoggerBinding.cs) it compares two values.

File: tests/ArmTemplates.Tests/Extractor/Scenarios/LoggerExtractorTests.cs [hunk 3]
```diff
@@ -70,5 +72,63 @@ public async Task GenerateLoggersTemplates_ProperlyLaysTheInformation()
...
+            foreach(var api in apisToTest)
+            {
+                await loggerExtractor.LoadAllReferencedLoggers(new List<string>() { api }, extractorParameters);
+            }
+
+            // assert
+            loggerExtractor.Cache.ServiceLevelDiagnosticLoggerBindings.Count.Should().Be(1);
+            loggerExtractor.Cache.ApiDiagnosticLoggerBindings.Count.Should().Be(2);
+            
+            foreach( var (key, value) in expectedApiDiagnosticLoggerBingingsValues)
```
[the remark this answers is not in the document]
Comment (f-alizada) [reply tests/ArmTemplates.Tests/Extractor/Scenarios/LoggerExtractorTests.cs]: Thank you, updated.

File: src/ArmTemplates/Extractor/EntityExtractors/LoggerExtractor.cs [hunk 4]
```diff
@@ -102,7 +102,7 @@ public class LoggerExtractor : ILoggerExtractor
             return loggerTemplate;
         }
 
-        async Task LoadAllReferencedLoggers(
+        public async Task LoadAllReferencedLoggers(
```
[the remark this answers is not in the document]
Comment (f-alizada) [reply src/ArmTemplates/Extractor/EntityExtractors/LoggerExtractor.cs]: Agree, removed and GenerateLoggerTemplateAsync method is called from tests.

File: src/ArmTemplates/Extractor/EntityExtractors/LoggerExtractor.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (DeagleGross) [reply src/ArmTemplates/Extractor/EntityExtractors/LoggerExtractor.cs:151]: thanks for explaining!

**Review by DeagleGross — APPROVED** [review]

**Pull request merged**